server/game/cards/07-GR/ImmortalGreking.js

```diff
@@ -0,0 +1,47 @@
+const Card = require('../../Card.js');
+
+class ImmortalGreking extends Card {
+    // Play: Take control of an enemy creature and place it anywhere
+    // in your battleline. Until Immortal Greking leaves play, that
+    // creature belong's to Immortal Greking's house.
+    //
+    // Destroyed: If you are haunted, archive Immortal Greking.
+    setupCardAbilities(ability) {
+        this.play({
+            target: {
+                cardType: 'creature',
+                controller: 'opponent',
+                gameAction: ability.actions.sequential([
+                    ability.actions.cardLastingEffect((context) => ({
+                        duration: 'lastingEffect',
+                        effect: [
+                            ability.effects.takeControl(context.source.controller),
+                            ability.effects.takeControlOnRight()
```

Comment:
I know putting it on the right first and then moving it in a separate action isn't perfect, but it didn't seem worth complicating the `takeControl` logic for deploying.  I can give it a try if you all feel strongly about it though.